test/Test/Database/LSMTree/Internal/Snapshot/FS.hs

```diff
@@ -137,3 +157,100 @@ instance Arbitrary TestErrors where
       [ TestErrors writeErrors' readErrors'
       | (writeErrors', readErrors') <- shrink (writeErrors, readErrors)
       ]
+
+{-------------------------------------------------------------------------------
+  Snapshot corruption
+-------------------------------------------------------------------------------}
+
+-- | A 'Double' in the @[0, 1)@ range.
+newtype Double_0_1 = Double_0_1 Double
+  deriving stock (Show, Eq)
+
+instance Arbitrary Double_0_1 where
+  arbitrary = Double_0_1 <$> genDouble
+  shrink (Double_0_1 x) = [Double_0_1 x' | x' <- shrink x, 0 <= x', x' < 1]
+
+prop_flipSnapshotBit ::
+     Positive (Small Int)
+  -> V.Vector (Word64, Entry Word64 Word64)
+  -> Double_0_1 -- ^ Used to pick which file to corrupt
```

Comment:
Isn't it possible to use an actual arbitrary choice over the set of files?

Reply:
The set of files depends on multiple factors, and it's hard to predict from the test inputs. This seemed to simplest way to pick from a list that you don't know the size of beforehand. Do you have an idea for how to approach it differently?

Reply:
Left a TODO on the property